JeMPI_Apps/JeMPI_UI/src/hooks/useAuth.tsx

```diff
@@ -120,36 +80,116 @@ export const AuthProvider = ({ children }: AuthProviderProps): JSX.Element => {
   useEffect(
     () => {
       if (config.useSso) {
-        if (!isLoading) {
-          if (!user && !isLoginPage) {
+        if (!authContext.isLoading) {
+          if (!authContext.user && !isLoginPage) {
             navigate({ pathname: '/login' })
-          } else if (user && isLoginPage) {
+          } else if (authContext.user && isLoginPage) {
             navigate({ pathname: '/' })
           }
         }
       }
     },
     // eslint-disable-next-line react-hooks/exhaustive-deps
-    [isLoading]
+    [authContext.isLoading] //TODO: Check
+  )
+
+  if (config.useSso && authContext.isLoading) {
+    return <LoadingSpinner id="user-loading-spinner" />
+  }
+
+  if (
+    config.useSso &&
+    !authContext.isLoading &&
+    !authContext.isAuthenticated &&
+    !isLoginPage
+  ) {
+    return (<React.Fragment>
+            {authContext.error && <ApiErrorMessage error={authContext.error} />}
+            <LoadingSpinner id="user-loading-spinner" />
+          </React.Fragment>)
+  }
+
+  return (
+    <React.Fragment>
+      {authContext.error && <ApiErrorMessage error={authContext.error} />}
+      {children}
+    </React.Fragment>
   )
+}
+
+export const AuthProvider = ({ children }: AuthProviderProps): JSX.Element => {
+  const queryClient = useQueryClient()
+  const { enqueueSnackbar } = useSnackbar()
+  const key = 'auth-user'
+
+  const {
+    data: user,
+    isLoading,
+    error,
+    refetch
+  } = useQuery<User, AxiosError<unknown, User>>({
+    queryKey: [key],
+    queryFn: async () => {
+      return await ApiClient.getCurrentUser()
+    },
+    retry: false,
+    refetchOnWindowFocus: false,
+    enabled: config.useSso
+  })
+
+  const { mutate: logout } = useMutation({
+    mutationFn: ApiClient.logout,
+    onSuccess(data: any, navigate: any) {
```

Comment:
Please avoid using `any` type

Reply:
Hi @issambaccouch . Have update this. Data can by any/has no fixed definition, and we are not using it for anything as well.